Next version

hansvancalster opened the pull request:
Branch to collect contributions for new version of the package.

Closes #13

falkmielke committed:
dhmv: adjust/extend `get_coverage_wcs()`*

- dhmv api support
- adjusted roxygen
- more assertions
- version constraint
- case-insensitive wcs

falkmielke committed:
dhmv: adjust/extend `get_coverage_wcs()`*

- dhmv api support
- adjusted roxygen
- more assertions
- version constraint
- case-insensitive wcs

falkmielke committed:
dhmv: vignette

falkmielke committed:
dhmv: vignette

hansvancalster committed:
bump version

hansvancalster committed:
update documentation

hansvancalster committed:
fix vignette (issue #13)

hansvancalster committed:
fix url

hansvancalster committed:
fix url

hansvancalster committed:
use II* as cue for start of geoTIFF

hansvancalster committed:
fix invalid url notes

falkmielke committed:
patch-mht: wfs/wcs vignette additions

falkmielke committed:
DHMV API: rebase, mht adjustment, vignette*

+ rebased to follow the latest dev by @hansvancalster on parallel
branches
+ incorporated review suggestions
+ modified `unpack_mht()`: the function will now return the tif path
+ additional vignette paragraph: using DHMV `v2.0.1`

falkmielke committed:
DHMV API: rebase led to merge conflicts

hansvancalster committed:
address warning about deprecated bootstrap 3

hansvancalster committed:
fix URLs

hansvancalster committed:
update dictionary

hansvancalster committed:
add NEWS item

hansvancalster committed:
fix URL

hansvancalster committed:
Merge pull request #15 from inbo/patch-mht

Patch mht

falkmielke committed:
dhmv api: update with `next-version`

falkmielke committed:
DHMV API: news entry + a bug

falkmielke committed:
DHMV API: author @falkmielke

falkmielke committed:
DHMV API: checklist spell checks and linting

falkmielke committed:
DHMV API: checklist adjustments

hansvancalster committed:
fix linters

hansvancalster committed:
fix spelling issues

hansvancalster committed:
fix urls and add ows4R to suggests

hansvancalster committed:
minor tweaks to vignette

hansvancalster committed:
add Falk as contributor

hansvancalster committed:
update citation

hansvancalster committed:
Merge pull request #12 from inbo/dhmv_api

add DHMV API

hansvancalster committed:
update citation

hansvancalster committed:
update pkgdown

hansvancalster committed:
update .Rbuildignore

hansvancalster committed:
update css

hansvancalster committed:
fix yml

hansvancalster committed:
also detect big endian byte order geoTIFF files

hansvancalster commented:
@falkmielke I will ask you to review this branch, but it is more a pro-forma question since most of the stuff here is already reviewed in the other branch. The only substantial addition is a small change in the unpack_mht function to also detect geoTIFF with different type of byte order.

@florisvdh would you like to review too? Or do you perhaps have something in mind to contribute? This would be a good time.

falkmielke reviewed (approved):
I could successfully install the `next-version` branch locally, vignettes build as expected, outcomes look fine. I have also looked at all code changes, seems good. Thank you for bringing together the new version!

For the archives: I have kept notes to better test the geoTIFF extraction (not tested now) as examples roll over my desk; I assume it worked for your cases. I also added to my todo list to create unit tests for the different wcs services, where we query known outcomes. In particular, you mentioned a strict CRS 31370 requirement, which I will double check once I understand more about that. 
Since the package works, and we practically only use 31370, these extras are not urgent. But let me know if we should keep track via an issue.

falkmielke committed:
tryCatch for KMI temperature query*

KMI query failed when remote-installing `next-release`
Wrapped in a tryCatch to ensure building of vignettes.

florisvdh commented on vignettes/spatial_dhmv_query.Rmd:
```diff
@@ -0,0 +1,490 @@
+| format              | image/tiff                        |
+| request             | GetCoverage                       |
+
+You can selectively shuffle and adjust components, generalizing the working query and learning about the WCS API. Which feels a bit like [fuzzing](https://en.wikipedia.org/wiki/Fuzzing) the `geo.api`.
```
'geo.api' confused me a bit here (hence so does the geo.api subdomain). I was confusing with the recent [OGC API](https://ogcapi.ogc.org/), i.e. the successor of OGC Web Service Standards.  IIUC, OGC API has dedicated server/client implementations in Java and Python, called the OGC "[GeoAPI](https://www.ogc.org/publications/standard/geoapi/)" standard, and is still evolving. The OGC API offers modern Web API approaches compared to the (legacy!) OGC Web Service Standards (which will probably remain in place for a long time).

Yet we shouldn't confuse with OGC API / GeoAPI since we're dealing with web services, not OGC API.

BTW, interacting with OGC API is provided by GDAL ([vector](https://gdal.org/en/latest/drivers/vector/oapif.html) / [raster](https://gdal.org/en/latest/drivers/raster/ogcapi.html)), which is the obvious gateway for R packages like {sf} and {terra}, as these already interface GDAL.

(Also, the webservice standards are supported by GDAL, hence should be able to use with GDAL-interfacing R packages)

The term 'geo.api' pops up a couple of times in the vignette so maybe you can improve or replace it (maybe 'WCS request' or something).

falkmielke replied:
Interesting and good to know! I have replaced all "API" by WCS or interfaces, except when writing about the `geo.api.vlaanderen.be`.

florisvdh replied:
Thanks!

florisvdh commented on R/get_coverage_wcs.R:
```diff
@@ -4,7 +4,7 @@
 #' from which it is read with `terra::rast()` - if needed reprojected -
 #' and returned as a `SpatRaster`object
```
```suggestion
#' and returned as a `SpatRaster` object
```

(This is actually not part of this PR)

florisvdh commented on R/get_coverage_wcs.R:
```diff
@@ -20,7 +20,8 @@
 #'   - `"dsm"`: digital surface model Flanders
-#' See [metadata Vlaanderen](https://metadata.vlaanderen.be/metadatacenter/srv/dut/catalog.search#/search?keyword=OGC:WCS) for more information # nolint: line_length_linter.
+#'   - `"dhmv"`: digital elevation model Flanders (contains dtm and dsm data)
+#' See [metadata Vlaanderen](https://metadata.vlaanderen.be/srv/eng/catalog.search#/search?any=WCS) for more information # nolint: line_length_linter.
```
The `# nolint: line_length_linter` string appears as-is in the resulting Rd file... I don't know whether roxygen comments have a way to support this, but the current result doesn't look so nice :confused:.

falkmielke replied:
I have rearranged that comment.

florisvdh reviewed (commented):
This review (more or less) adds to the review of PR #12.

@falkmielke I've read the spatial_dhmv_query vignette. Pleasant reading, relatively easy to follow, well done! :sun_with_face: 

I locally tested the `get_coverage_wcs()` example in the vignette, with success!

Only minor comments added.

A minor general one is my advice to put each sentence of a paragraph in another line. Hugo will stitch those lines into one paragraph in HTML, as would Pandoc. It minimizes git diffs and it will also make review comments easier to interpret afterwards. No need to implement it here as far as I'm concerned, just saying!

Don't forget to rebuild the documentation (Rd files) after making updates in the roxygen comment section.

falkmielke commented on vignettes/spatial_dhmv_query.Rmd:
```diff
@@ -0,0 +1,490 @@
+Note that the *capabilities* xml holds a lot of valuable information:
+
+-   First and foremost, search it for `CoverageID`; in this case, I was interested in `DHMVII_DTM_1m`: grid coverage of the LIDAR elevation data for Flanders.
+-   We can also find that the Coordinate Reference System (CRS) `crsSupported` is EPSG:31370, aka. [Belgian Lambert 72](https://epsg.io/31370). Always good to know.
```
Thank you for the pointers, which I have read and stored. 
I will adjust and include the dynamic CRS naming.

falkmielke commented on vignettes/spatial_dhmv_query.Rmd:
```diff
@@ -0,0 +1,490 @@
+
+However, this is also a story of tracing errors in existing tools, and scraping other tools for hints.
+
+Can we actually trace the error back to a source? It was pure co-incidence that QGIS provided a working link, or are there general strategies?
```
Thank you for confirming; I took the liberty of copying most of this comment into the vignette, it seems worth re-iterating the praise of QGIS.

falkmielke committed:
next-version: incorporate comments by @florisvdh

falkmielke committed:
next-version: documentation update

hansvancalster committed:
link to epsg.org instead of epsg.io

hansvancalster committed:
fix spelling

hansvancalster committed:
use error = TRUE chunk option instead of trycatch()

- wrapping in trycatch is adding code which will confuse novice users

hansvancalster committed:
make sure vignettes are shown on website ;-)

hansvancalster committed:
better nav bar text color

hansvancalster committed:
fix colour css for right side bare navigation

hansvancalster committed:
add url

hansvancalster committed:
update documentation

hansvancalster committed:
update citation

hansvancalster committed:
Merge pull request #16 from inbo/next-version-check

minor additions to incorporate review feedback
